launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,266 @@
+    String mp3Link = "https://storage.googleapis.com/" + BUCKET_NAME + "/" + entityId;
+
+    // Create embedded entity to store MP3 data
+    EmbeddedEntity mp3 = new EmbeddedEntity();
+    mp3.setProperty(ID, entityId);
+    mp3.setProperty(MP3_LINK, mp3Link);
+    mp3.setProperty(EMAIL, email);
+
+    // Retrieve entity from Datastore
+    Entity savedEntity;
+    try {
+      savedEntity = datastore.get(entityKey);
+    } catch(EntityNotFoundException e) {
+      throw new IOException("Entity not found in Datastore.");
+    }
+
+    // Generate xml string
+    RSS rssFeed = new RSS(podcastTitle, description, language, email, mp3Link);
+    String xmlString = "";
+    try {
+      xmlString = RSS.toXmlString(rssFeed);
+      savedEntity.setProperty(XML_STRING ,xmlString);
+    } catch(IOException e){
+      throw new IOException("Unable to create XML string.");
+    }
+
+    // Update entity by adding embedded mp3 entity as a property
+    savedEntity.setProperty(MP3, mp3);
+    datastore.put(savedEntity);
+
+    //write the file upload form
+    String formHtml = generateSignedPostPolicyV4(PROJECT_ID, BUCKET_NAME, entityId);
+    res.setContentType("text/html");
+    res.getWriter().println(formHtml);
+  }
+
+  /**
+   * Generate policy for directly uploading a file to Cloud Storage via HTML form.
+   * @return HTML form (as a String) for uploading MP3
+   */
+  public String generateSignedPostPolicyV4 (String projectId, String bucketName, String blobName) throws IOException {
+    // String blobName = entity ID from Datastore, as the name of object uploaded to GCS
+
+    String errorMessage = "Policy is null.";
+    Storage storage;
+    PostPolicyV4.PostFieldsV4 fields;
+    PostPolicyV4 policy = null;
+    String myRedirectUrl = "";
+    try {
+      // Prepare credentials and API key
+      String keyFileName = "launchpod-step18-2020-47434aafba88.json";
+      ClassLoader classLoader = getClass().getClassLoader();
+      File file = new File(classLoader.getResource(keyFileName).getFile());
+      GoogleCredentials credentials = GoogleCredentials.fromStream(new FileInputStream(file));
+
+      storage = StorageOptions.newBuilder().setCredentials(credentials).build().getService();
+
+      myRedirectUrl="https://launchpod-step18-2020.appspot.com/rss-feed?action=generateRSSLink&id=" + blobName;
+      fields =
+          PostPolicyV4.PostFieldsV4.newBuilder().setSuccessActionRedirect(myRedirectUrl).build();
+
+      policy =
+        storage.generateSignedPostPolicyV4(
+            BlobInfo.newBuilder(bucketName, blobName).build(), 10, TimeUnit.MINUTES, fields);
+    } catch(Exception e) {
+      errorMessage = e.getMessage();
+      throw e;
+    }
+
+    StringBuilder htmlForm;
+    if (policy!=null) {
+      htmlForm =
+          new StringBuilder(
+              "<form name='mp3-upload' action='"
+                  + policy.getUrl()
+                  + "' method='POST' enctype='multipart/form-data'>\n");
```

Comment:
use String.format() here instead of using + 

also use String.format for below